sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/MutableColumnarRow.java

```diff
@@ -39,17 +38,10 @@
  */
 public final class MutableColumnarRow extends InternalRow {
   public int rowId;
-  private final ColumnVector[] columns;
-  private final WritableColumnVector[] writableColumns;
-
-  public MutableColumnarRow(ColumnVector[] columns) {
-    this.columns = columns;
-    this.writableColumns = null;
-  }
+  private final WritableColumnVector[] columns;
```

Comment:
Could this extend `ColumnarBatchRow` to avoid duplication?

Reply:
I thought about it too. `MutableColumnarRow` is used in performance critical path (hash aggregate), and I'm a little hesitant to add class Hierarchy here, which may hurt performance.  cc @kiszk